semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala

```diff
@@ -226,7 +232,12 @@ trait SymbolOps { self: SemanticdbOps =>
       sym.isSyntheticValueClassCompanion ||
       sym.isUselessField ||
       sym.isSyntheticCaseAccessor ||
-      sym.isRefinementClass
+      sym.isRefinementClass ||
+      sym.isSyntheticJavaModule
+    }
+    def isUselessOccurrence: Boolean = {
+      sym.isUseless &&
+      !sym.isSyntheticJavaModule
     }
```

Comment:
Why separate this from `isUseless`?

Reply:
I added a comment
```
      !sym.isSyntheticJavaModule // references to static Java inner classes should have occurrences
```
Without `isUselessOccurrence` we would miss the occurrence to `Map` in `java.util.Map.Entry` because the reference points to a synthetic Java module symbol.